File: v-lang/l-java/v-spring/v-web-interceptor/src/main/java/org/example/config/WebMvcConfig.java
package org.example.config;

import jakarta.annotation.Resource;
import org.example.controller.interceptor.FooInterceptor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebMvcConfig implements WebMvcConfigurer {

    @Resource
    private FooInterceptor fooInterceptor;

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(fooInterceptor)
                .addPathPatterns("/**")
                .excludePathPatterns("/ping");
    }
}

File: tmp-archives/v-reproduce/protect-api/src/main/kotlin/org/example/config/WebMvcConfig.kt
package org.example.config

import org.example.controller.interceptor.SignInterceptor
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebMvcConfig : WebMvcConfigurer {

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(SignInterceptor())
            .addPathPatterns("/**")
            .excludePathPatterns("/ping")
    }
}